src/parser/parser_support.c

```diff
@@ -709,6 +709,47 @@ pt_is_expr_wrapped_function (PARSER_CONTEXT * parser, const PT_NODE * node)
   return false;
 }
 
+/*
+ * pt_is_json_function () -
+ *   return: true if node is a json function
+ *   parser(in): parser context
+ *   node(in): PT_FUNTION node
+ */
+bool
+pt_is_json_function (PARSER_CONTEXT * parser, const PT_NODE * node)
+{
+  FUNC_TYPE function_type;
+
+  if (node->node_type == PT_FUNCTION)
+    {
+      function_type = node->info.function.function_type;
+      if (function_type == F_JSON_ARRAY
+	  || function_type == F_JSON_ARRAY_APPEND || function_type == F_JSON_ARRAY_INSERT
+	  || function_type == F_JSON_CONTAINS || function_type == F_JSON_CONTAINS_PATH
+	  || function_type == F_JSON_DEPTH
+	  || function_type == F_JSON_EXTRACT
+	  || function_type == F_JSON_GET_ALL_PATHS
+	  || function_type == F_JSON_INSERT
+	  || function_type == F_JSON_KEYS
+	  || function_type == F_JSON_LENGTH
+	  || function_type == F_JSON_MERGE || function_type == F_JSON_MERGE_PATCH
+	  || function_type == F_JSON_OBJECT
+	  || function_type == F_JSON_PRETTY
+	  || function_type == F_JSON_QUOTE
+	  || function_type == F_JSON_REMOVE
+	  || function_type == F_JSON_REPLACE
+	  || function_type == F_JSON_SEARCH
+	  || function_type == F_JSON_SET
+	  || function_type == F_JSON_TYPE || function_type == F_JSON_UNQUOTE || function_type == F_JSON_VALID
+	  || function_type == PT_JSON_ARRAYAGG || function_type == PT_JSON_OBJECTAGG)
```

Comment:
how about rewrite like "if (function_type >= F_JSON_OBJECT and function_type <= F_JSON_VALID)"

Reply:
If you check many if conditions without branching, a switch statement like below might be more efficient.
```

switch(node->info.function.function_type)
{
     case F_JSON_ARRAY:
     case F_JSON_ARRAY_APPEND:
     case: F_JSON_ARRAY_INSERT:
    ...
	case PT_JSON_OBJECTAGG:
               return true;
         default:
               break;
} 
```

Reply:
Okay. "if (function_type >= F_JSON_OBJECT and function_type <= F_JSON_VALID)" is good.

Reply:
PT_JSON_ARRAYAGG, PT_JSON_OBJECTAGG This is not possible because of these two. I'll change it to a switch